### ISSUE TITLE
Font-size

### ISSUE THREAD

**Issue by agzam** [stumpwm/stumpwm#783]
Total noob here. Please help. I just installed Stumpwm and I can't figure out one thing. I just need to change the size of the font, it's too small on my laptop. For the panel that appears when `which-key-mode` is on (what's the name of it?). I've been trying all sort of things. 

First I tried the approach described in https://wiki.archlinux.org/index.php/Stumpwm, and it would fail on `(ql:quickload "clx-truetype")`. After many attempts of adding/removing dependencies; cloning the repo and building it manually; re-installing stump from AUR, etc., I learned that `clx-truetype` was removed from quick-lisp repo(s). I have no idea why, and I don't know if there's a way to get it installed locally anyway.

So then I tried searching the manual. I've found `set-font` function, it doesn't say how you use it, and neither the docstring in the source code gives you any hints, so I search a bit more and found this: https://github.com/stumpwm/stumpwm/blob/master/sample-stumpwmrc.lisp#L40
`(set-font "-xos4-terminus-medium-r-normal--14-140-72-72-c-80-iso8859-15")`
I have no idea how stump parses the string, what format it should be and how to change it to a different font. I tried adding that to my rcfile, changed `14` to `24` and did `loadrc` in stumpish - and got no results. 

Can someone please help me. I would love to get a nicer font there like "JetBrains Mono", but seems ttf fonts currently have some issues, can you at least show me a way to change the size of the default, bitmap font? Thanks!

**Comment by PuercoPop** [comment]
> learned that clx-truetype was removed from quick-lisp repo(s). I have no idea why, and I don't know if there's a way to get it installed locally anyway.

Yes, you can get it from here: https://github.com/LispLima/clx-truetype. Clone it under `~/quicklisp/local-projects/`. Then you should be able to use JetBrains Mono. I recommend you go this route.

```lisp
(asdf:make "ttf-fonts")
;; run (clx-truetype:cache-fonts) once or everytime you change your system fonts.
(set-font (make-instance 'xft:font
                          :family "JetBrains Mono"
                          :subfamily "Regular"
                          :size 12
                          :antialias t))
```

> what format it should be and how to change it to a different font.

The terminus example you mention uses the [X Logical Font Description](https://wiki.archlinux.org/index.php/X_Logical_Font_Description).

**Comment by agzam** [comment]
I am now able to load `(ql:quickload "clx-truetype")`, but for some reason it can't locate the fonts I have. I have number of fonts in `/usr/share/fonts`, but for none of them `make-instance` works. What can I try?

**Comment by agzam** [comment]
Finally, I was able to make it work.

	;; git clone git@github.com:LispLima/clx-truetype.git ~/quicklisp/local-projects/clx-truetype
	(ql:quickload "clx-truetype")
	(load-module "ttf-fonts")
        
        ;; next thing is important (nothing else has worked for me) 
	(clx-truetype:cache-font-file "/usr/share/fonts/TTF/JetBrainsMono-Medium.ttf")

	(set-font
	 (make-instance 'xft:font
			:family "JetBrains Mono"
			:subfamily "Regular"
			:size 16
			:antialias t))

I spent two days trying to get it right. Please don't get me wrong, I'm not in a position of asking for something or even worse, of making demands, I'm just a newcomer user. I'm simply sharing my thoughts.

I think Stumpwm is incredible and has a lot of potential, but stuff like this absolutely does hurt the adoption. As someone who completely sold on the ideas implemented in Stumpwm (and frankly, I am not seeing anyone building a better alternative), I do now care about its future.

Authors and maintainers, please consider either making the default font option nicer and more readable, or simplify the procedure of setting a custom font. Thanks!

**Comment by PuercoPop** [comment]
> (clx-truetype:cache-font-file "/usr/share/fonts/TTF/JetBrainsMono-Medium.ttf")

So `(clx-truetype:cache-fonts)` didn't work for you? I think it didn't work for one other user but in theory it should have found JetBrainsMono.

> I'm simply sharing my thoughts.

It is healthy that you do so. That is what the issue tracker is for after all.

> stuff like this absolutely does hurt the adoption.

I agree.

**Comment by dmb2** [comment]
Hi @agzam,

Welcome!  I'm so sorry you encountered one of our more annoying friction points so soon.

You are correct, the default font is both tiny and hard to read.  This is because StumpWM uses CLX to draw graphics primitives independent of modern toolkits.  This made sense when StumpWM was originally written, and it reduced dependencies, but it has not aged well.  

I'm so glad you got it worked out! I wish it wasn't so difficult to get working out of the box.  I've tried a number of different ways to improve font rendering including building my own renderer, but for now we're stuck with CLX out of the box.  

We do our best to make things as easy as possible to transition, the managing paradigm stumpwm uses is very unique and very difficult to get started with.  Thanks for the feedback!  I hope we can get good, stable, easy to use font rendering in the future.

**Comment by agzam** [comment]
> I'm so sorry you encountered one of our more annoying friction points so soon.

Oh, no worries. I learned quite a few important concepts while struggling with that. I realized how cool Common Lisp is, I decided to learn it. Yes, just for the sake of configuring StumpWM.

To be honest, I felt a bit frustrated, and wanted to give up, I'm glad I didn't. Now I'm quickly progressing on discovering some amazing features of StumpWM - I'm remapping the keys in different apps, learning contrib modules, etc. I'm hoping at some point I'll learn enough to be able to contribute. StumpWM is awesome and quite unique in its class. I am completely sold and will be spreading the word of how cool it is. Thank you everyone for making my life a little bit less frustrating.
